asset_stream_manager/main.cc

```diff
@@ -67,18 +67,26 @@ absl::Status Run(const AssetStreamConfig& cfg) {
 }
 
 std::string GetLogPath(const char* log_dir, const char* log_base_name) {
-  DefaultSystemClock clock;
-  std::string timestamp_ext = clock.FormatNow(".%Y%m%d-%H%M%S.log", false);
+  DefaultSystemClock* clock = DefaultSystemClock::GetInstance();
+  std::string timestamp_ext = clock->FormatNow(".%Y%m%d-%H%M%S.log", false);
   return path::Join(log_dir, log_base_name + timestamp_ext);
 }
 
-void InitLogging(const char* log_dir, bool log_to_stdout, int verbosity) {
+void InitLogging(std::string& log_dir, bool log_to_stdout, int verbosity) {
   LogLevel level = cdc_ft::Log::VerbosityToLogLevel(verbosity);
   if (log_to_stdout) {
     cdc_ft::Log::Initialize(std::make_unique<cdc_ft::ConsoleLog>(level));
   } else {
-    cdc_ft::Log::Initialize(std::make_unique<cdc_ft::FileLog>(
-        level, GetLogPath(log_dir, "assets_stream_manager").c_str()));
+    if (path::ExpandPathVariables(&log_dir).ok() &&
+        path::CreateDirRec(log_dir).ok()) {
+      cdc_ft::Log::Initialize(std::make_unique<cdc_ft::FileLog>(
+          level, GetLogPath(log_dir.c_str(), "assets_stream_manager").c_str()));
+    } else {
+      LOG_WARNING("Failed to create log directory or expand directory path %s",
+                  log_dir);
+      cdc_ft::Log::Initialize(std::make_unique<cdc_ft::ConsoleLog>(level));
```

Comment:
I think it's better to stop with an error at this point instead of creating a console log. If the process is started in the background, there would be no visible feedback for the user that the log dir is wrong.